Add contract compilation & collection docs (#2064)
foundry-rs/starknet-foundry: docs/src/testing/contracts-collection.md

Reviewer: You have to add this to the `SUMMARY.md` file, otherwise it is not visible. Please always inspect docs 🙏 
```
cd docs & mdbook serve --open
```

### How Contracts Are Collected

> When you call `snforge test`, one of the things that Forge does is that it calls Scarb, particularly `scarb build`.

Reviewer: Forge -> `snforge`

It makes Scarb build all contracts from your package and save them to the `target/{current_profile}` directory
(read more on [Scarb website](https://docs.swmansion.com/scarb/docs/extensions/starknet/contract-target.html)).

> Then, Foundry loads compiled contracts from each package (for all packages in the project) and uses them to run tests.

Reviewer: ```suggestion
Then, `snforge` loads compiled contracts from the package your tests are in, enabling you to declare the contracts in tests.
```

> ⚠️ **Warning**
>
> Make sure to define `[[target.starknet-contract]]` section in your `Scarb.toml`, otherwise Scarb won't build your contracts.

## Using External Contracts In Tests

> If you wish to use external contracts inside your tests (e.g. an ERC20 token, an account contract) from your dependencies,

Reviewer: ```suggestion
If you wish to use contracts from your dependencies inside your tests (e.g. an ERC20 token, an account contract),
```
